### ISSUE TITLE
Compile the pdf documentation and report on error

### ISSUE THREAD

**Issue by nthiery** [robertwb/sage-patchbot#64]
I recently wasted some good release manager time by setting a positive review on a ticket where the pdf documentation actually did not build. It would be a nice addition if the patchbot would try to build  the pdf documentation and report on the result. I grepped for `pdf` in the patchbot sources and did not find any match, so this is not the case now,  right?

A difficulty might be that building the pdf doc requires quite a few dependencies (pdflatex, many packages, ...). So maybe this should be optional.

Thanks!

**Comment by fchapoton** [comment]
Yes, this is not the case now.
The bot always run "make doc-clean" then "make", so it builds the html doc
I have started to prepare a plugin "docbuild_pdf" that will run "make doc-pdf"
This has not been tested yet. It will certainly not be activated by default.

**Comment by fchapoton** [comment]
Which ticket was it ? What was the error ?

**Comment by nthiery** [comment]
See trac.sagemath.org/ticket/16659#comment:98.
Thanks!

**Comment by fchapoton** [comment]
There exists a plugin "docbuild_pdf", that is not activated by default. I think nobody has tried to use it.

**Comment by videlec** [comment]
Sadly it is not that simple to allow an extra plugin... the point is that the list of plugins should appear in a given order and "docbuild_pdf" should appear right after the "docbuild_html". The only way I know to achieve that is to copy/paste the list from patchbot.py into my configuration file with the "docbuild_pdf" inserted at the right place. It would be easier with a more extensible configuration with

- order or priorities for the plugins (with default as it is currently in patchbot.py where "docbuild_pdf" appears)
- list of allowed plugins (with default as it is currently in patchbot.py, i.e. "docbuild_pdf" disallowed)
- list of extra allowed plugins (empty by default but can be added "dobuild_pdf")

What do you think?

**Comment by fchapoton** [comment]
Je ne comprend pas. Have you tried just to enable the doc_pdf plugin on one of your bot using your json config file? It should not matter whether or not you build the pdf doc before or after the html doc, no ?

**Comment by videlec** [comment]
The order of plugins does matter... anyway how do you enable the doc_pdf plugin?

**Comment by fchapoton** [comment]
In principle, you just write it inside your custom plugin list, in the json
config file.

2016-08-04 18:50 GMT+02:00 Vincent Delecroix <notifications@github.com>:

**Comment by videlec** [comment]
Looking at the code: the entry "plugins" in the configuration is the list of **all**  plugins. In other words, I do not see how to enable the doc pdf plugin without copy/paste from the code.

**Comment by fchapoton** [comment]
yes, indeed. But this offers the possibility to add and remove all plugins
at will. The docbuild one is forced in any case because otherwise doctests
are failing.

2016-08-04 20:09 GMT+02:00 Vincent Delecroix <notifications@github.com>:

**Comment by videlec** [comment]
Hence you can reread my previous message which propose an alternative implementation with more options. We can even follow more standard terminology with

- "plugins_available" : the list of available plugins. The order will be respected at execution.
- "plugins_enabled": the list of extra plugins to enable (a list of extra plugin to be added to the default ones)
- "plugins_disabled": the list of plugins to disable (a list of plugins to disable from the default ones)

**Comment by fchapoton** [comment]
ok, indeed. Feel free to implement something like that, but please keep
things simple.

2016-08-04 20:21 GMT+02:00 Vincent Delecroix <notifications@github.com>:

**Comment by videlec** [comment]
now sagemath/sage-patchbot#91

**Comment by mkoeppe** [comment]
For the list of needed latex packages, see https://trac.sagemath.org/ticket/31529, which needs help
